go/vt/vtgate/engine/fk_cascade.go

```diff
@@ -82,77 +97,149 @@ func (fkc *FkCascade) TryExecute(ctx context.Context, vcursor VCursor, bindVars
+			// We check if the column has updated or not.
+			isUnchanged, err := row[info.CompExprCol].ToBool()
+			if err != nil {
+				return err
+			}
+			if !isUnchanged {
+				// If any column has changed, then we can't skip this row.
+				// We need to execute the child primitive.
+				skipRow = false
+				break
+			}
+		}
+		// If none of the columns have changed, then there is no update to cascade, we can move on.
+		if skipRow {
+			continue
+		}
+		// We create a bindVariable that stores the tuple of columns involved in the fk constraint.
 		bv := &querypb.BindVariable{
 			Type: querypb.Type_TUPLE,
 		}
-		for _, row := range selectionRes.Rows {
-			var tupleValues []sqltypes.Value
-			for _, colIdx := range child.Cols {
-				tupleValues = append(tupleValues, row[colIdx])
-			}
-			bv.Values = append(bv.Values, sqltypes.TupleToProto(tupleValues))
+		// Create a tuple from the Row.
+		var tupleValues []sqltypes.Value
+		for _, colIdx := range child.Cols {
+			tupleValues = append(tupleValues, row[colIdx])
 		}
+		bv.Values = append(bv.Values, sqltypes.TupleToProto(tupleValues))
 		// Execute the child primitive, and bail out incase of failure.
 		// Since this Primitive is always executed in a transaction, the changes should
-		// be rolled back incase of an error.
+		// be rolled back in case of an error.
 		bindVars[child.BVName] = bv
-		_, err = vcursor.ExecutePrimitive(ctx, child.Exec, bindVars, wantfields)
+
+		// Next, we need to copy the updated expressions value into the bind variables map.
+		for _, info := range child.NonLiteralInfo {
+			bindVars[info.UpdateExprBvName] = sqltypes.ValueBindVariable(row[info.UpdateExprCol])
+		}
+		var err error
+		if isStreaming {
+			err = vcursor.StreamExecutePrimitive(ctx, child.Exec, bindVars, wantfields, func(result *sqltypes.Result) error { return nil })
+		} else {
+			_, err = vcursor.ExecutePrimitive(ctx, child.Exec, bindVars, wantfields)
+		}
 		if err != nil {
-			return nil, err
+			return err
 		}
+		// Remove the bind variables that have been used and are no longer required.
 		delete(bindVars, child.BVName)
+		for _, info := range child.NonLiteralInfo {
+			delete(bindVars, info.UpdateExprBvName)
+		}
 	}
-
-	// All the children are modified successfully, we can now execute the Parent Primitive.
-	return vcursor.ExecutePrimitive(ctx, fkc.Parent, bindVars, wantfields)
+	return nil
 }
 
 // TryStreamExecute implements the Primitive interface.
 func (fkc *FkCascade) TryStreamExecute(ctx context.Context, vcursor VCursor, bindVars map[string]*querypb.BindVariable, wantfields bool, callback func(*sqltypes.Result) error) error {
-	// We create a bindVariable for each Child
-	// that stores the tuple of columns involved in the fk constraint.
-	var bindVariables []*querypb.BindVariable
-	for range fkc.Children {
-		bindVariables = append(bindVariables, &querypb.BindVariable{
-			Type: querypb.Type_TUPLE,
-		})
-	}
-
+	var selectionRes *sqltypes.Result
```

Comment:
I suggest this method simply calls the `TryExecute` method in this file and then feeds the result to the visitor. Safer and simpler to just do it sequentially instead

Reply:
Okay 👍